internal/serverinstall/docker.go

```diff
@@ -14,14 +14,25 @@ import (
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/jsonmessage"
 	"github.com/docker/go-connections/nat"
+
+	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
 	"github.com/hashicorp/waypoint/internal/clicontext"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
 	pb "github.com/hashicorp/waypoint/internal/server/gen"
 	"github.com/hashicorp/waypoint/internal/serverconfig"
 )
 
-func InstallDocker(
-	ctx context.Context, ui terminal.UI, scfg *Config) (
+type DockerInstaller struct {
+	Config DockerConfig
+}
+
+type DockerConfig struct {
```

Comment:
I think we can lowercase all these and keep them unexported (same with Nomad/K8S).